Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUExternalTextureDescriptor.h

```diff
@@ -27,15 +27,29 @@
 
 #include "WebGPUObjectDescriptorBase.h"
 #include "WebGPUPredefinedColorSpace.h"
+#include <wtf/Vector.h>
 
 typedef struct __CVBuffer* CVPixelBufferRef;
 
 namespace PAL::WebGPU {
 
+struct HTMLVideoElementIdentifier {
+    uint64_t identifier;
+};
+struct WebCodecsVideoFrameIdentifier {
+    std::pair<uint64_t, uint64_t> identifier;
+};
+
+using VideoSourceIdentifier = std::variant<HTMLVideoElementIdentifier, WebCodecsVideoFrameIdentifier>;
+
 struct ExternalTextureDescriptor : public ObjectDescriptorBase {
-    uint64_t mediaIdentifier { 0 };
+    VideoSourceIdentifier mediaIdentifier;
     PredefinedColorSpace colorSpace { PredefinedColorSpace::SRGB };
+#if PLATFORM(COCOA)
+    RetainPtr<CVPixelBufferRef> pixelBuffer { nullptr };
+#else
     CVPixelBufferRef pixelBuffer { nullptr };
```

Comment:
This doesn't make a lot of sense. Why should there be be a `CVFoo` type in `!PLATFORM(COCOA)`?

Reply:
It is to make WinCairo compile without adding `#if ENABLE(VIDEO) && PLATFORM(COCOA)` in a few places.

But indeed it should probably not be here. I will add `#if ENABLE(VIDEO) && PLATFORM(COCOA)` in a few places instead